### PR TITLE
updates for support of Jena 2.13.0 (apache-jena-libs 2.13.0)

### PR DESCRIPTION
**Pull request by m1ci**

### CODE REVIEW

**Comment by RubenVerborgh** [comment]
Thanks, will look closely and merge.

File: src/main/java/org/linkeddatafragments/solver/ReorderTransformationLDF.java [hunk 1]
```diff
@@ -61,7 +61,6 @@ private void initializeMatcher() {
         matcher.addPattern(new Pattern(numTriples, VAR, VAR, VAR));     // ???
     }
 
-    @Override
```
Comment (RubenVerborgh) [inline src/main/java/org/linkeddatafragments/solver/ReorderTransformationLDF.java:64]: We shouldn't lose this, should we?

File: src/main/java/org/linkeddatafragments/model/LinkedDataFragmentGraph.java [hunk 2]
```diff
@@ -175,4 +175,9 @@ public static String toString( String prefix, Graph that )
         return b.toString();
     }
 
+    @Override
+    protected ExtendedIterator<Triple> graphBaseFind(Triple triple) {
+        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
```
Comment (RubenVerborgh) [inline src/main/java/org/linkeddatafragments/model/LinkedDataFragmentGraph.java:180]: I propose

``` Java
return find(triple);
```

Any objections to that?

Comment (m1ci) [inline src/main/java/org/linkeddatafragments/model/LinkedDataFragmentGraph.java:180]: sure why not.

File: src/main/java/org/linkeddatafragments/solver/ReorderTransformationLDF.java (hunk 1 above)
Comment (m1ci) [inline src/main/java/org/linkeddatafragments/solver/ReorderTransformationLDF.java:64]: in 2.13.0 there is no ReorderTransformationBase but the class was renamed to ReorderTransformationSubstitution

Comment (RubenVerborgh) [inline src/main/java/org/linkeddatafragments/solver/ReorderTransformationLDF.java:64]: Alright, but the `weight` remains, or not? (If not, the method doesn't have a purpose and should perhaps be removed.)

**Comment by RubenVerborgh** [comment]
Alright, so the one thing stopping me from merging this is [the `@Override` on `weight`](https://github.com/LinkedDataFragments/Client.Java/pull/6/files#diff-2e8a176c50d641f96b7c5845f366352bL64).

Are we overriding a `weight` method in the base class or not?
If we are, then `@Override` must be there.

If we are not, then `@Override` must not be there, but then the entire `weight` method becomes pointless, because nothing will ever call it.

Could you help to clarify this @m1ci or @egonw?

**Comment by RubenVerborgh** [comment]
Given that [`weight is still there in 2.13`](https://github.com/apache/jena/blob/jena-2.13.0/jena-arq/src/main/java/com/hp/hpl/jena/sparql/engine/optimizer/reorder/ReorderTransformationSubstitution.java#L235), we must keep `@Override`. If no objections, I'll change and merge like that.

File: src/main/java/org/linkeddatafragments/solver/ReorderTransformationLDF.java (hunk 1 above)
Comment (egonw) [inline src/main/java/org/linkeddatafragments/solver/ReorderTransformationLDF.java:64]: Good point... it is gone, but then it is not clear when this method would be used... do you have a test which relies on the function of this method? But honestly, I have no clue at this moment how the new class replaces they weight functionality...

Comment (RubenVerborgh) [inline src/main/java/org/linkeddatafragments/solver/ReorderTransformationLDF.java:64]: In the same way, it seems. [Only the class name was changed.](https://github.com/apache/jena/commit/31a36d32d238ca803097f6769367745ed06c7b51)

**Comment by egonw** [comment]
Confirmed... still there, so, yes, overwrite... note, I don't think the annotation is required for it to actually overwrite the method... so, removing the method annotation should still overwrite the method, AFAIK.

**Comment by RubenVerborgh** [comment]
It's indeed not required, but it prevents us from making mistakes.

**Comment by egonw** [comment]
Agreed! But I don't understand why it gave a compile error, if the method was still there... maybe this has changed in Java8? Are you compiling with Java8 too? Maybe this is related: http://stackoverflow.com/a/27369217/217943, and maybe Java8 is more strict?

**Comment by RubenVerborgh** [comment]
Can you try to see if you have errors again?
Maybe your compilation environment wasn't fully initialized yet?

**Comment by RubenVerborgh** [comment]
Merged as afddd7613833a4fea1a50b3ce40180d573177950.

**Comment by egonw** [comment]
OK, done testing. The current master compiles fine for me.

**Comment by mielvds** [comment]
Yes, java 8 is more strict on this, override is required, or at least the default compiler setting says so.

**Pull request closed without merging**